src/libraries/Microsoft.PowerFx.Interpreter/Functions/Mutation/PatchFunction.cs

```diff
@@ -52,6 +53,18 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
             return isValid;
         }
 
+        public override void CheckSemantics(TexlBinding binding, TexlNode[] args, DType[] argTypes, IErrorContainer errors)
+        {
+            base.CheckSemantics(binding, args, argTypes, errors);
+            if (!binding.IsMutable(args[0]))
+            {
+                errors.EnsureError(
+                    args[0],
+                    new ErrorResourceKey("ErrorResource_MutationFunctionCannotBeUsedWithImmutableValue"),
+                    this.Name);
+            }
+        }
```

Comment:
can we share with Collect? 


Does Set() need this check? Here's what it does today:
[https://github.com/microsoft/Power-Fx/blob/813e341e3996b12312500bbf31ce869549a3039e/src/libraries/Microsoft.PowerFx.Interpreter/Functions/SetFunction.cs#L72](https://github.com/microsoft/Power-Fx/blob/813e341e3996b12312500bbf31ce869549a3039e/src/libraries/Microsoft.PowerFx.Interpreter/Functions/SetFunction.cs#L72)

Reply:
Set can also be used to define a variable, so the check would be different in that function.

Reply:
2 issues here:
1. Sharing with Collect + Patch  
2. Set - in Fx, the variable is already defined by time we call this. For analysis in PA, I get we have an earlier phase that creates definitions. 


Can we do 1 here?

Reply:
Updating...